include/sparrow/typed_array.hpp

```diff
@@ -0,0 +1,358 @@
+         */
+        reference front();
+        const_reference front() const;
+        ///@}
+
+        ///@{
+        /*
+         * @brief Access the last element.
+         *
+         * @description Returns a reference to the last element.
+         * Calling back on an empty container is undefined.
+         *
+         * @return A reference to the last element.
+         */
+        reference back();
+        const_reference back() const;
+        ///@}
+
+        // Iterators
+
+        ///@{
+        /*
+         * @brief Returns an iterator to the beginning.
+         *
+         * @description Returns an iterator to the first element of the vector.
+         * If the vector is empty, the returned iterator will be equal to end().
+         *
+         * @return An iterator to the beginning.
+         */
+        iterator begin();
+        const_iterator cbegin() const;
+        ///@}
+
+        ///@{
+        /**
+         * @brief Returns an iterator to the end.
+         *
+         * @description Returns an iterator to the element following the last element of the vector.
+         * This element acts as a placeholder; attempting to access it results in undefined behavior.
+         *
+         * @return An iterator to the end.
+         */
+        iterator end();
+        const_iterator cend() const;
+        ///@}
+
+        /*
+         * @brief Returns a range of the bitmap.
+         *
+         * @return A range of the bitmap.
+         */
+        const_bitmap_range bitmap() const;
+
+        /*
+         * @brief Returns a range of the values.
+         *
+         * @return A range of the values.
+         */
+        const_value_range values() const;
+
+        // Capacity
+
+        /*
+         * @brief Checks whether the container is empty.
+         *
+         * @description Returns whether the container is empty (i.e. whether its size is 0).
+         *
+         * @return true if the container is empty, false otherwise.
+         */
+        bool empty() const;
+
+        /*
+         * @brief Returns the number of elements.
+         *
+         * @description Returns the number of elements in the container, i.e. std::distance(begin(), end()).
+         *
+         * @return The number of elements in the container.
+         */
+        size_type size() const;
+
+        // TODO: Add reserve, capacity, shrink_to_fit
+
+        // Modifiers
+
+        // TODO: Implement insert, erase, push_back, pop_back, clear, resize, swap
+
+        std::partial_ordering operator<=>(const typed_array<T>& other) const;
+
+        bool operator==(const typed_array<T>& other) const;
+
+        bool operator!=(const typed_array<T>& other) const;
+
+        bool operator<(const typed_array<T>& other) const;
+
+        bool operator>(const typed_array<T>& other) const;
+
+        bool operator<=(const typed_array<T>& other) const;
+
+        bool operator>=(const typed_array<T>& other) const;
```

Comment:
In C++20, these four operators are automatically generated when you implement `operator<=>`